app/tests/utils.py

```diff
@@ -226,3 +227,18 @@ def recurse_callbacks(callbacks):
 
     if new_callbacks:
         recurse_callbacks(callbacks=new_callbacks)
+
+
+@contextlib.contextmanager
+def playwright_trace(context, directory=None, name="trace"):
+    directory.mkdir(parents=True, exist_ok=True)
```

Comment:
Won't this leave a bunch of test files on disk? Why not use a temporary directory in a context handler?

Reply:
I will leave this for Chris to comment on and potentially change since he added this, but if this is a temporary file, it will be gone when the test has run and then we can't open it after, right?

Reply:
Yes but you could run the tests with `--pdb --lf` and get a breakpoint on the exception with access to the files.

Reply:
ATM is a debug insertion to get a trace. The traces need to be viewed via a GUI; the quickest way is to dump them somewhere where I can directly read them. (Without going through the hassle of doing a docker copy; running these tests take long enough already.)

Reply:
We can also just get rid of it. It was just for debugging purposes anyway.

Reply:
Yes, generating traces is not something we'd like to do all the time, anyway.